### PR TITLE
fix: Remove rxjs events to avoid duplicate custom events in Angular

### PR DESCRIPTION
**Pull request by ethanWallace**
# Summary | Résumé

With the `v0.22.0` release of the `@cdssnc/gcds-components-angular`, `@Outputs` were added to the generated angular components. Unfortunately this was now causing custom events like `gcdsClick` to be emitted twice.

To fix, I removed Stencil's implementation of events from `rxjs` to remove the duplicate emitted events.

## Example angular component

```
import {Component} from '@angular/core';

@Component({
  selector: 'app-click-me-gcds',
  template: ` <gcds-button type="button" (gcdsClick)="onClickMe()">Click me!</gcds-button>
    <gcds-pagination type="list" current-page="3" total-pages="20" (gcdsClick)="onClick($event)"></gcds-pagination>
`,
})
export class ClickMeGcdsComponent {
  clickMessage = '';

  onClickMe() {
    console.log('clicked')
  }
  onClick(event: any) {
    event.preventDefault();
    console.log(event.detail)
  }
}
```

Previously the `gcdsClick` would be emitted twice when either the button or pagination links were clicked. Now only one event is emitted.

### CODE REVIEW

File: utils/angular-output-target/src/generate-angular-component.ts [hunk 1]
```diff
@@ -51,6 +51,10 @@ export const createAngularComponentDefinition = (
     proxyCmpOptions.push(`\n  methods: [${formattedMethods}]`);
   }
 
+  /**
+   * Custom modification
+   * - Added conditional to add outputs
+   */
```
Comment (melaniebmn) [inline utils/angular-output-target/src/generate-angular-component.ts:57]: Thanks for adding extra comments :)

**Review by melaniebmn — DISMISSED** [review]
LGTM

**Comment by daine** [comment]
FYI I'm still testing this -- should get to it by tonight :)

File: packages/angular/src/lib/stencil-generated/angular-component-lib/utils.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 /* eslint-disable */
 /* tslint:disable */
-import { fromEvent } from 'rxjs';
+/**
+ * Custom modification
+ * - Replaced import { fromEvent } from 'rxjs'; with import { EventEmitter } from '@angular/core';
+ */
```
Comment (daine) [inline packages/angular/src/lib/stencil-generated/angular-component-lib/utils.ts:6]:
```suggestion
/**
 * GCDS Modification
 * Replaced 
 *     import { fromEvent } from 'rxjs'; 
 * From: https://github.com/ionic-team/stencil-ds-output-targets/blob/9524c1ce970770e01afb493c292f71a2fe61b14a/packages/angular-output-target/angular-component-lib/utils.ts#L3
 */
```

File: packages/angular/src/lib/stencil-generated/angular-component-lib/utils.ts [hunk 3]
```diff
@@ -34,8 +38,12 @@ export const proxyMethods = (Cmp: any, methods: string[]) => {
   });
 };
 
-export const proxyOutputs = (instance: any, el: any, events: string[]) => {
-  events.forEach((eventName) => (instance[eventName] = fromEvent(el, eventName)));
+/**
+ * Custom modification
+ * - Replaced fromEvent from rxjs package with EventEmitter from @angular/core
+ */
```
Comment (daine) [inline packages/angular/src/lib/stencil-generated/angular-component-lib/utils.ts:44]:
```suggestion
/**
 * GCDS modification
 * Replaced 
 *   fromEvent(el, eventName)
 * to
 *   new EventEmitter()
 * From: https://github.com/ionic-team/stencil-ds-output-targets/blob/9524c1ce970770e01afb493c292f71a2fe61b14a/packages/angular-output-target/angular-component-lib/utils.ts#L38
 */
```

File: utils/angular-output-target/angular-component-lib/utils.ts [hunk 4]
```diff
@@ -1,6 +1,10 @@
 /* eslint-disable */
 /* tslint:disable */
-import { fromEvent } from 'rxjs';
+/**
+ * Custom modification
+ * - Replaced import { fromEvent } from 'rxjs'; with import { EventEmitter } from '@angular/core';
+ */
```
Comment (daine) [inline utils/angular-output-target/angular-component-lib/utils.ts:6]: Oops I commented on the generated file first sorry!

```suggestion
/**
 * GCDS Modification
 * Replaced 
 *     import { fromEvent } from 'rxjs'; 
 * From: https://github.com/ionic-team/stencil-ds-output-targets/blob/9524c1ce970770e01afb493c292f71a2fe61b14a/packages/angular-output-target/angular-component-lib/utils.ts#L3
 */
```

File: utils/angular-output-target/angular-component-lib/utils.ts [hunk 5]
```diff
@@ -34,8 +38,12 @@ export const proxyMethods = (Cmp: any, methods: string[]) => {
   });
 };
 
-export const proxyOutputs = (instance: any, el: any, events: string[]) => {
-  events.forEach((eventName) => (instance[eventName] = fromEvent(el, eventName)));
+/**
+ * Custom modification
+ * - Replaced fromEvent from rxjs package with EventEmitter from @angular/core
+ */
```
Comment (daine) [inline utils/angular-output-target/angular-component-lib/utils.ts:44]:
```suggestion
/**
 * GCDS modification
 * Replaced 
 *   fromEvent(el, eventName)
 * to
 *   new EventEmitter()
 * From: https://github.com/ionic-team/stencil-ds-output-targets/blob/9524c1ce970770e01afb493c292f71a2fe61b14a/packages/angular-output-target/angular-component-lib/utils.ts#L38
 */
```

File: utils/angular-output-target/src/generate-angular-component.ts (hunk 1 above)
Comment (daine) [inline utils/angular-output-target/src/generate-angular-component.ts:57]:
```suggestion
  /**
   * GCDS modification
   * - Added conditional to add outputs
   */
```

File: utils/angular-output-target/src/generate-angular-component.ts [hunk 6]
```diff
@@ -67,6 +71,10 @@ export const createAngularComponentDefinition = (
    * Angular does not complain about the inputs property. The output target
    * uses the inputs property to define the inputs of the component instead of
    * having to use the @Input decorator (and manually define the type and default value).
+   * 
+   * Custom modifications
+   * - Add hasOutputs logic to render outputs
+   * - Modify proxyOutouts to remove second parameter "this.el"
```
Comment (daine) [inline utils/angular-output-target/src/generate-angular-component.ts:77]:
```suggestion
   * 
   * GCDS modifications
   * - Added `hasOutputs` logic to render outputs
   * - Modified proxyOutputs to remove second parameter `this.el`
   *      proxyOutputs(this, this.el, [${formattedOutputs}]);`
   *   From: https://github.com/ionic-team/stencil-ds-output-targets/blob/9524c1ce970770e01afb493c292f71a2fe61b14a/packages/angular-output-target/src/generate-angular-component.ts#L82
```

**Review by daine — CHANGES_REQUESTED** [review]
Works pretty well, I tested by using our npm package 0.22.1 to verify that the events were being emitted twice, and then tested this PR and I can confirm that it is no longer firing twice.

Hoping to add more info in the comments about the changes so I added some suggestions in line. Thanks @ethanWallace !

**Pull request merged**